### DISCUSSION TITLE
Automated API Documentation for GraphQL

### DISCUSSION THREAD

**Discussion by zbhavyai** [quarkusio/quarkus#25382] [Q&A] [answered]
Hello,

The guide available on [https://quarkus.io/guides/openapi-swaggerui](https://quarkus.io/guides/openapi-swaggerui) explains how Smallrye OpenAPI extension can be used to generate REST API specification documentation.

Is there a similar setup available for GraphQL API as well? Or for GraphQL only schema is available at `<url:port>/graphql/schema.graphql`?

Thank you.

[a comment by an automated account is not shown]

**Reply by phillip-kruger** [reply]
Is this what you are after : https://quarkus.io/guides/smallrye-graphql#ui ? This is the "swagger-ui" for GraphQL called graphiql (see https://github.com/graphql/graphiql)

**Reply by zbhavyai** [reply]
I was thinking like pages generated by openapi-swaggerui or [https://github.com/2fd/graphdoc](https://github.com/2fd/graphdoc), which probably can be shared easily.

But yes, I had explored graphiql, and it has a similar feature set in terms of playing around the with API interactively.

**Reply by phillip-kruger** [reply]
Out of the box we only support graphiql. You should be able to add graphoc yourself as any quarkus app can also serve static resource

**Reply by zbhavyai** [reply]
Sure. Thank you so much for your response! Would try it if needed be.
